### ISSUE TITLE
`prefer-builtin-properties-and-methods`

### ISSUE THREAD

**Issue by fisker** [vuejs/eslint-plugin-vue#2567]
**Please describe what the rule should do:**

**What category should the rule belong to?**

[x] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<!-- Bad -->
<script setup>
const emit = defineEmits('foo')
</script>
<template>
	<button @click="emit('foo')">Foo</button>
</template>
```

```vue
<!-- Good -->
<script setup>
const emit = defineEmits('foo')
</script>
<template>
	<button @click="$emit('foo')">Foo</button>
</template>
```

```vue
<!-- Bad -->
<script setup>
import { useSlots, useAttrs } from 'vue'

const slots = useSlots()
</script>
<template>
	<div>
		<div v-if="slots.foo"><slot name="foo"></slot></div>
	</div>
</template>
```

```vue
<!-- Good -->
<script setup>
import { useSlots, useAttrs } from 'vue'

const slots = useSlots()
</script>
<template>
	<div>
		<div v-if="$slots.foo"><slot name="foo"></slot></div>
	</div>
</template>
```

**Additional context**

It can also apply to [other properties and methods](https://vuejs.org/api/component-instance.html)

**Comment by ota-meshi** [comment]
Thank you for proposing the rule. That rule sounds good to me.
But some users may prefer the opposite style. I think it would be better if it was an option.
What do you think?

**Comment by jacekkarczmarczyk** [comment]
Definitely, there should be a possibility to set `always` or `never`
